src/driver.rs

```diff
@@ -410,6 +410,87 @@ impl DriverManager {
         Ok(Driver { c_driver })
     }
 
+    /// Get the Driver based on the file extension from filename
+    ///
+    /// Searches for the available extensions in the registered
+    /// drivers and returns the matches. The determined driver is
+    /// checked for writing capabilities as
+    /// [`Dataset::open`](Dataset::open) can already open datasets
+    /// with just path.
+    ///
+    /// See also: [`get_driver_by_name`](Self::get_driver_by_name)
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// let compatible_drivers =
+    ///     DriverManager::guess_drivers_for_write("test.gpkg", true)
+    ///         .map(|d| d.short_name())
+    ///         .collect::<Vec<String>>();
+    /// println!("{:?}", compatible_drivers);
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// ```text
+    /// ["GPKG"]
+    /// ```
+    pub fn guess_drivers_for_write(
+        filename: &str,
+        is_vector: bool,
```

Comment:
An enum would probably be nicer here.

Reply:
Enum for Vector/Raster? I was thinking if we do that, we should either use the OpenOptions that has those already, but also has other options like read/write flag we can look at. But it has a lot of variants and adding checks for all of them would be complicated.

And if we're making new enum, should we just do vector and raster, or should we add, raster multiband and other raster formats, as well as vector specific subtypes as well. All of those seem complicated. Hence simple bool as we're only looking at vector or raster.

Reply:
> we should either use the OpenOptions

I'm not sure. `GDAL_OF_XXX_BLOCK_ACCESS` we certainly don't want, the others might be nice to have. A separate enum is certainly more clear and I'm starting to like the idea.

> And if we're making new enum, should we just do vector and raster, or should we add, raster multiband and other raster formats, as well as vector specific subtypes as well. All of those seem complicated. Hence simple bool as we're only looking at vector or raster.

It's fine, we can start with what the function supports. The problem with a bool is that you see `guess_driver("x.nc", true)` and you have no idea what `true` means unless you check. `guess_driver("x.nc", FormatOptions::Raster)` and you immediately know what's up.

Reply:
I made a simple enum with raster and vector variants for now. We can add more things as we add other stuffs.